### DISCUSSION TITLE
esp32s3 host bare_api build issues

### DISCUSSION THREAD

**Discussion by j105rob** [hathach/tinyusb#1733] [Q&A]
I have a esp32s3 devkitc board. My host computer is running Ubuntu 22.04. I am able to build and flash the cdc_msc_freertos example without any issues.

When I try to build the host/bare_api (or any other host examples), I get several errors. The first one is that the code to set $TOP in the cmake file is missing for these examples. I was able to fix that error by copying into the CMakeLists.txt the relevant lines from the cdc_msc_freertos/CMakeLists.txt. 

After I get past that error, I am presented with another error which I can't figure out. The output I am seeing from the build process is:

CMake Error at /home/ubuntu/esp/esp-idf/tools/cmake/build.cmake:263 (message):
  Failed to resolve component 'src'.

The function that is failing and producing the above output is supposed to "Resolve the requirement component to the component target created for that component". I am not sure what this means. I have tried comparing the make files between the 2 example dirs and I feel like I am just going down rabbit holes.

Can anyone guide me in the right direction? Can someone else try to build the example for the FAMILY=esp32s3 and BOARD=espressif_s3_devkitc to see if the error can be reproduced?

Any help is appreciated! TIA!

**Comment by hathach** [comment]
host mode is not supported on esp32s2/3 just yet https://github.com/hathach/tinyusb/blob/master/docs/reference/supported.rst
